tests/js/legendTests.js

```diff
@@ -0,0 +1,299 @@
+                args: ["{that}.dom.row", "mouseover", "gpii.tests.chartAuthoring.mouseOverListener"]
+            }
+        },
+        members: {
+            mouseOverListenerCalled: false
+        }
+    });
+
+    gpii.tests.chartAuthoring.objectArray = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayAdd = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id4",
+        value: 26,
+        label: "Five"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayRemove = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayChangeInPlace = [{
+        id: "id0",
+        value: 36,
+        label: "I"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "II"
+    }, {
+        id: "id2",
+        value: 26,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.objectArraySorted = [{
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id0",
+        value: 15,
+        label: "One"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayAddSorted = [{
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id4",
+        value: 26,
+        label: "Five"
+    },
+    {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id0",
+        value: 15,
+        label: "One"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayRemoveSorted = [{
+        id: "id1",
+        value: 67,
+        label: "Two"
+    },
+    {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id0",
+        value: 15,
+        label: "One"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayChangeInPlaceSorted = [{
+        id: "id1",
+        value: 67,
+        label: "II"
+    }, {
+        id: "id0",
+        value: 36,
+        label: "I"
+    }, {
+        id: "id2",
+        value: 26,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.mouseOverListener = function (data, i, that) {
+        that.mouseOverListenerCalled = true;
```

Comment:
Do you care how many times it was called? If so, it may be better to have a counter instead of a boolean value.